modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceAdapter.java

```diff
@@ -261,6 +275,21 @@ public Source getDelegate() {
     return source;
   }
 
+  public SourceTransactionalAction getTransactionalAction() throws MuleException {
+    ValueResolver valueResolver = nonCallbackParameters.getResolvers().get(TRANSACTIONAL_ACTION_PARAMETER_NAME);
+
+    if (valueResolver.isDynamic()) {
```

Comment:
this seems overly defensive or you're hiding a bug elsewhere. You should not worry about this at this point since the parameter model doesn't accept expressions. If you ever got a dynamic resolver here, then that's a bug you need to investigate